File: src/main/java/bot/util/database/SqlFunctions.java
package bot.util.database;

import bot.enums.Language;
import org.postgresql.util.PSQLException;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;

public class SqlFunctions {

    private final Connection connection;

    public SqlFunctions(Connection connection) {
        this.connection = connection;
    }

    public boolean insertLanguage(Long chatId, Language language){
        try {
            PreparedStatement preparedStatement = connection.prepareStatement(SqlScripts.INSERT_LANGUAGE);

            preparedStatement.setLong(1, chatId);
            preparedStatement.setString(2, language.toString());

            preparedStatement.execute();

            return true;
        } catch (SQLException throwables) {
            throwables.printStackTrace();
            return false;
        }
    }

    public Language selectLanguage(Long chatId){
        try {
            PreparedStatement preparedStatement = connection.prepareStatement(SqlScripts.SELECT_LANGUAGE);

            preparedStatement.setLong(1, chatId);

            ResultSet resultSet = preparedStatement.executeQuery();

            try {
                resultSet.next();
                String language = resultSet.getString("language");
                return Language.convert(language);
            } catch (PSQLException e){
                boolean insert = insertLanguage(chatId, Language.EN);
                if (insert){
                    return Language.EN;
                }
                else return null;
            }

        } catch (SQLException throwables) {
            throwables.printStackTrace();
            return null;
        }
    }
}

File: src/main/java/bot/util/database/SqlFunctions.kt
package bot.util.database

import bot.enums.Language
import bot.util.Log
import org.postgresql.util.PSQLException
import java.sql.Connection
import java.sql.SQLException

class SqlFunctions(private val connection: Connection?) {
    private val log = Log()


    fun insertLanguage(chatId: String, language: Language): Boolean {
        return try {
            val preparedStatement = connection!!.prepareStatement(SqlScripts.INSERT_LANGUAGE)
            preparedStatement.setLong(1, chatId.toLong())
            preparedStatement.setString(2, language.toString())
            preparedStatement.execute()
            true
        } catch (throwables: SQLException) {
            log.userException(throwables)
            false
        }
    }

    fun updateLanguage(chatId: String, language: Language): Boolean {
        return try {
            val preparedStatement = connection!!.prepareStatement(SqlScripts.UPDATE_LANGUAGE)
            preparedStatement.setLong(2, chatId.toLong())
            preparedStatement.setString(1, language.toString())
            preparedStatement.executeUpdate()
            true
        } catch (throwables: SQLException) {
            log.userException(throwables)
            false
        }
    }

    fun selectLanguage(chatId: String): Language? {
        return try {
            val preparedStatement = connection!!.prepareStatement(SqlScripts.SELECT_LANGUAGE)
            preparedStatement.setLong(1, chatId.toLong())
            val resultSet = preparedStatement.executeQuery()
            try {
                resultSet.next()
                val language = resultSet.getString("language")
                Language.convert(language)
            } catch (e: PSQLException) {
                val insert = insertLanguage(chatId, Language.EN)
                if (insert) {
                    Language.EN
                } else null
            }
        } catch (throwables: SQLException) {
            log.userException(throwables)
            null
        }
    }
}